rhythm/extendedparser.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright (c) 2024 Diagrid Inc.
+Licensed under the MIT License.
+*/
+
+// Forked from https://github.com/dapr/kit/
+// Original License:
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package time contains utilities for working with times, dates, and durations.
+//
+//nolint:nakedret,nonamedreturns
+package rhythm
+
+import (
+	"errors"
+	"strconv"
+	"time"
+)
+
+func ParseISO8601Duration(from string) (years int, months int, days int, duration time.Duration, repetition int, err error) {
+	// -1 signifies infinite repetition
+	repetition = -1
+
+	// Length must be at least 2 characters per specs
+	l := len(from)
+	if l < 2 {
+		err = errors.New("unsupported ISO8601 duration format: " + from)
+		return
+	}
+
+	var i int
+
+	// Check if the first character is "R", indicating we have repetitions
+	if from[0] == 'R' {
+		// Scan until the "/" character to get the repetitions
+		for {
+			i++
+			if i == l || from[i] == '/' {
+				break
+			}
+		}
+
+		if i-1 < 1 {
```

Comment:
`i < 2`

Reply:
This is copied from Dapr. I did not want to change at this point.